lib/phlex/compiler/method_compiler.rb

```diff
@@ -165,6 +177,23 @@ def compile_plain_helper(node)
 			end
 		end
 
+		def compile_fragment_helper(node)
+			node.copy(
+				block: compile_fragment_helper_block(node.block)
+			)
+		end
+
+		def compile_fragment_helper_block(node)
+			node.copy(
+				body: [
+					statement("__phlex_original_should_render__ = __phlex_should_render__"),
+					statement("__phlex_should_render__ = __phlex_state__.should_render?;"),
+					compile_block_body_node(node.body),
+					statement("__phlex_should_render__ = __phlex_original_should_render__"),
```

Comment:
Because the local is first created at the top of the method, we need to put it back after changing it.